test/e2e/validation_test.go

```diff
@@ -91,8 +93,44 @@ func testUserData(t *testing.T) {
 	require.NoError(t, err, "error creating signer using private key")
 	pubKeyBytes := ssh.MarshalAuthorizedKey(signer.PublicKey())
 	require.NotNil(t, pubKeyBytes, "failed to retrieve public key using signer for private key")
-	found := &corev1.Secret{}
+	found := &core.Secret{}
 	err = framework.Global.Client.Get(context.TODO(), kubeTypes.NamespacedName{Name: "windows-user-data", Namespace: "openshift-machine-api"}, found)
-	require.NoError(t, err, "could not find windows user data secret in required namespace")
+	require.NoError(t, err, "could not find Windows user data secret in required namespace")
 	assert.Contains(t, string(found.Data["userData"][:]), string(pubKeyBytes[:]), "expected user data not present in required namespace")
 }
+
+// testUserDataTamper tests if userData reverts to previous value if updated
+func testUserDataTamper(t *testing.T) {
+	secretInstance := &core.Secret{}
+	validUserDataSecret, err := framework.Global.KubeClient.CoreV1().Secrets("openshift-machine-api").Get(context.TODO(), "windows-user-data", meta.GetOptions{})
+	require.NoError(t, err, "could not find Windows userData secret in required namespace")
+
+	var tests = []struct {
+		name           string
+		operation      string
+		expectedSecret *core.Secret
+	}{
+		{"Update the userData secret with invalid data", "Update", validUserDataSecret},
+		{"Delete the userData secret", "Delete", validUserDataSecret},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.operation == "Update" {
+				updatedSecret := validUserDataSecret.DeepCopy()
+				updatedSecret.Data["userData"] = []byte("invalid data")
+				err := framework.Global.Client.Update(context.TODO(), updatedSecret)
+				require.NoError(t, err, "could not update userData secret")
+			}
+			if tt.operation == "Delete" {
+				err := framework.Global.KubeClient.CoreV1().Secrets("openshift-machine-api").Delete(context.TODO(), "windows-user-data", meta.DeleteOptions{})
+				require.NoError(t, err, "could not delete userData secret")
+			}
+
+			err = framework.Global.Client.Get(context.TODO(), kubeTypes.NamespacedName{Name: "windows-user-data",
```

Comment:
I think you need to have wait with a timeout logic here as the apiserver update can take some time.